libs/SalesforceSDK/src/com/salesforce/androidsdk/security/PasscodeManager.java

```diff
@@ -511,22 +518,23 @@ public void setMinPasscodeLength(int minPasscodeLength) {
              * triggered later from OAuthWebviewHelper.
              */
             if (hasStoredPasscode(SalesforceSDKManager.getInstance().getAppContext())) {
-        		showLockActivity(SalesforceSDKManager.getInstance().getAppContext(),
-        				true);
+                setEnabled(true);
+                showLockActivity(ctx, true);
+                return true;
```

Comment:
Yeah - this stuff is quite convoluted. That's why I have avoided changing code here since I wrote it years ago ;-)